### PR TITLE
Fixes #33378 - Make pages layout styles consistent

### PR DESCRIPTION
**Pull request by Ron-Lavi**
Different pages are using different paddings and styles,
let's make the layout of the pages more consistent.

### CODE REVIEW

**Review by MariaAga — DISMISSED** [review]
Tested and it looks good and consistent! Also works good with Tasks plugin

**Review by amirfefer — COMMENTED** [review]
Looks like this change breaks some current styling (new host details page, katello)

**Comment by Ron-Lavi** [comment]
@amirfefer can you share the screenshots of the pages you found? I saw something but I think it's very minor, did you spot something radical?
I would suggest merging it as we still have time until 3.1-stable for small fixes.

**Comment by jeremylenz** [comment]
@laviro Please rebase to current develop, otherwise it seems to break all my settings when I switch branches  :confounded:

**Comment by jeremylenz** [comment]
I'm seeing an odd gray bar of padding at the top of the screen that I don't see on develop.

padding_top

**Comment by jeremylenz** [comment]
It seems to be adding too much whitespace at the top.  Not sure if that's something we need to fix here or in katello..

CV details: develop:
cv_detail_develop

CV details: this branch:
cv_detail_branch

CV page: develop:
cv_page_develop

CV page: this branch:
cv_page_branch

**Comment by MariSvirik** [comment]
There should be 16 px from the top to the breadcrumbs
Screenshot 2021-10-08 at 14 15 21

or if that's an index page, then 24px from the top to the page title

Screenshot 2021-10-08 at 14 17 10

See [Marvel](https://marvelapp.com/prototype/8813483/screen/80289006) for reference

**Comment by jeremylenz** [comment]
@laviro If you can fix it here, I feel like it makes sense to do that since the layout in Katello looks okay in develop.

**Comment by jeremylenz** [comment]
@laviro any updates here?

**Comment by Ron-Lavi** [comment]
Thanks, let's revisit this after branching and see if we need additional PR in Katello to properly unify this

**Comment by Ron-Lavi** [comment]
waiting a bit more for plugins to be stable.

**Comment by Ron-Lavi** [comment]
I think I resolved all issues, can you take another look? @jeremylenz @amirfefer @MariaAga 
@jeremylenz looks like CV page will need to drop the `grid-with-margin` once we will merge this PR, that class adds the additional 20px margin on the top.

**Comment by jeremylenz** [comment]
> looks like CV page will need to drop the grid-with-margin once we will merge this PR, that class adds the additional 20px margin on the top.

cc @Andrewgdewar

File: app/assets/stylesheets/base.scss [hunk 1]
```diff
@@ -11,6 +11,21 @@ body {
...
+  margin-bottom: 8px;
+  font-family: 'RedHatDisplay';
+  font-weight: 400;
+}
+
+#rails-app-content {
+  padding: 0 24px;
+
+  #breadcrumb {
+    margin-top: -8px;
```
Comment (Andrewgdewar) [inline app/assets/stylesheets/base.scss:25]: Avoid negative pixels wherever possible, not sure if this is avoidable or not.

File: app/assets/stylesheets/base.scss [hunk 2]
```diff
@@ -11,6 +11,21 @@ body {
...
 }
 
+h1 {
+  margin-top: 0px;
+  margin-bottom: 8px;
+  font-family: 'RedHatDisplay';
+  font-weight: 400;
+}
+
+#rails-app-content {
```
Comment (Andrewgdewar) [inline app/assets/stylesheets/base.scss:21]: This class causes spacing issues on every katello page

**Comment by Andrewgdewar** [comment]
Apologies, after testing this PR with my changes, I would recommend you close it for now and try adding in components of it after my two PR's are merged.

This breaks spacing on far too many katello pages, and likely will cause issues in every other plugin that inherits our react-layouts. 

The good thing is, this is only 30 lines of code. 
So you can likely recreate parts of this without too much effort.

This was also kind of bad timing, as I had two similar PR's in flight.

[katello](https://github.com/Katello/katello/pull/9849)
[foreman](https://github.com/theforeman/foreman/pull/8996)

File: app/views/layouts/base.html.erb [hunk 3]
```diff
@@ -57,7 +57,7 @@
       <% end %>
       <div
         id="rails-app-content"
-        class="container-fluid container-pf-nav-pf-vertical nav-pf-persistent-secondary"
```
Comment (Andrewgdewar) [inline app/views/layouts/base.html.erb:60]: This is another cause of issues, as we target the container-fluid in a few places. 
If this is something that you can't live without we can open a separate PR to amend that in Katello.

File: webpack/assets/javascripts/react_app/components/Layout/components/LayoutContainer.js [hunk 4]
```diff
@@ -2,7 +2,8 @@ import React, { useEffect } from 'react';
 import PropTypes from 'prop-types';
 
 const LayoutContainer = ({ isCollapsed, children }) => {
-  const classes = 'react-container container-fluid nav-pf-persistent-secondary';
```
Comment (Andrewgdewar) [inline webpack/assets/javascripts/react_app/components/Layout/components/LayoutContainer.js:5]: Same as above

**Comment by Ron-Lavi** [comment]
Thanks @Andrewgdewar,
as we discussed, let's merge yours first, and then I will retry to apply the styling - moving this back to draft.

**Comment by jeremylenz** [comment]
@Ron-Lavi Andrew's PR's are merged now, so this is ready for another update :)

**Comment by jeremylenz** [comment]
@Ron-Lavi what's the status here?

**Comment by Ron-Lavi** [comment]
Hey @jeremylenz! Didn't have time for it though I might start working at it after 3.2 is out

**Comment by jeremylenz** [comment]
@Ron-Lavi any updates?

**Comment by Andrewgdewar** [comment]
Is there enough code here to keep this alive? You could likely just close this and move it's related ticket to the backlog until you are ready to pick it up again.

**Comment by jeremylenz** [comment]
I concur with @Andrewgdewar. I'm going to remove this from our review queue for now.

**Comment by Ron-Lavi** [comment]
Moved it to draft, but I can close this one for now 🤣 
sorry, had no much time to work on it

**Pull request closed without merging**